### PR TITLE
PHP 8.1 support

### PR DESCRIPTION
**Pull request by lilHermit**
- Fixed the PHP 8.1 deprecation of passing non-string to ctype_digit
- Updated testsuite to CakePHP 4.3+

**Refs**

 https://wiki.php.net/rfc/deprecations_php_8_1#ctype_function_family_accepts_int_parameters
 https://www.php.net/manual/en/function.ctype-digit.php

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -21,7 +21,7 @@
         "source": "https://github.com/cakephp/acl"
     },
     "require": {
-        "php": "^7.2",
+        "php": ">=7.4.0",
         "cakephp/cakephp": "^4.0",
```
Comment (markstory) [inline composer.json:25]: Should we pin this to `^4.3` ?

**Review by markstory — COMMENTED** [review]
Looks good to me.

File: composer.json (hunk 1 above)
Comment (lilHermit) [reply composer.json:25]: @markstory  You mean `cakephp/cakephp ^4.3` ?

Comment (lilHermit) [reply composer.json:25]: The only reason I didn't was because only the tests need CakePHP 4.3, but that's really a decision for you guys

Comment (markstory) [reply composer.json:25]: Doesn't changing the min PHP version effectively bump up the cake version as well though?

Comment (lilHermit) [reply composer.json:25]: I've rebased the PHP version to `>=7.2` so we don't force a cakephp version bump

**Pull request merged**